ui/app/src/components/CreateDashboardDialog/CreateDashboardDialog.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+
+export interface CreateDashboardDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onSubmit: Dispatch<string>;
+  projectName: string;
+}
+
+/**
+ * Dialog used to create a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.projectName The project where the dashboard will be created.
+ * @constructor
+ */
+const CreateDashboardDialog = (props: CreateDashboardDialogProps) => {
+  const { open, onClose, onSubmit, projectName } = props;
+  const [name, setName] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    if (!e.target.value) {
+      setError('Required');
+    } else if (!/^[a-zA-Z0-9_.:-]+$/.exec(e.target.value)) {
+      setError('Allowed special characters are _ . : -');
```

Comment:
what do you think about allowing spaces and then automatically convert it to PascalCase internally?

Reply:
+1 to doing something like this, I was a little thrown off when I tried to first create a dashboard and by reflex wanted to type _Test Dashboard_ with a space

Reply:
Sure